### DISCUSSION TITLE
[Uczymy się architektury] 🚀🚀 Architektura w Azure - Messaging #azure-solutions-architect

### DISCUSSION THREAD

**Discussion by aryll** [AzureCommunityPL/group#189] [Wydarzenia]
Cześć! 🙂
Nowy rok - nowy temat do nauki architektury w Azure! Na następnym spotkaniu spróbujemy poznać różnice między różnymi serwisami przesyłania komunikatów oraz jak je łączyć w jedną architekturę!
Skupimy się na takich technologiach jak:
- Azure Service Bus,
- Azure Event Grid,
- Azure Event Hubs,
- Storage Queues oraz
- Azure Relay.

**Następne spotkanie 21/01/2021 godzina 20:00**

Oczywiście czekamy na Wasze 𝐝𝐢𝐚𝐠𝐫𝐚𝐦𝐲 tutaj: https://bit.ly/2LjDOlM

🚀🚀🚀 Zadanie 🚀🚀🚀
Zaprojektuj architekturę dla systemu rezerwacji biletów. Płatności muszą być obsługiwane przez system postawiony na on-premises (ze względów prawnych nie można go przenieść do chmury), reszta systemu może być hostowana w chmurze.
Założenia:
- komunikacja powinna odbywać się 𝐚𝐬𝐲𝐧𝐜𝐡𝐫𝐨𝐧𝐢𝐜𝐳𝐧𝐢𝐞
- w ramach niepoprawnej płatności użytkownik powinien być powiadomiony 𝐦𝐚𝐢𝐥𝐨𝐰𝐨
- ilość użytkowników aplikacji będzie na poziomie 10000 dziennie
- wielkość pojedynczego żądania jest na poziomie 100 kb
- ze względu na ograniczoną ilość biletów ważna jest kolejność zgłoszeń
- każda 𝐳𝐦𝐢𝐚𝐧𝐚 𝐳𝐚𝐬𝐨𝐛𝐮 w Azure powinna być zgłaszana mailowo do osoby odpowiedzialnej za audyt zasobów Azure
- każde wyświetlenie danego biletu powinno być 𝐩𝐫𝐳𝐞𝐜𝐡𝐨𝐰𝐲𝐰𝐚𝐧𝐞 w dowolnej formie

🤔 Jakich mechanizmów integracyjnych Azure byś użył i dlaczego?
Czekamy na Wasze diagramy i wyceny tutaj: https://bit.ly/2LjDOlM

➡️ Przydatne linki
- https://docs.microsoft.com/.../compare-messaging-services
- https://docs.microsoft.com/.../service-bus-azure-and...

💻 Link do spotkania: https://bit.ly/358gJcV
📅 Kalendarz: https://calndr.link/event/4w53WG3g2Y
📖 Poprzednie spotkania (wraz z nagraniami): https://azurecommunity.pl/

Masz pytanie? Napisz do mnie lub @maciejgos !

O cyklu spotkań **#azure-solution-architect:**
Uczymy się tworzyć architekturę w Azure! Jeśli myślisz, że to też Twoja ścieżka - dołącz do nas! Zapraszamy osoby o każdym poziomie zaawansowania!
Każde spotkanie to nowe architektoniczne zadanie, które pozwala nam poznać dany fragment Azure - omawiamy nasze propozycje i w ten sposób zdobywamy wiedzę! Bazujemy na wiedzy wymaganej na egzaminach AZ-303 oraz AZ-304, ale głównie skupiamy się na bardziej praktycznych i życiowych problemach.

**Comment by szympulka** [comment]
@aryll linki z ms nie działają :)

**Reply by maciejgos** [reply]
Dzięki za info już sprawdzam

**Reply by maciejgos** [reply]
Prawidłowe linki
- https://docs.microsoft.com/en-us/azure/event-grid/compare-messaging-services
- https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-azure-and-service-bus-queues-compared-contrasted

**Comment by phaate** [comment]
To ja moze pierwszy sprobuje. To nie jest ostateczna wersja :)
![image](https://user-images.githubusercontent.com/55980809/104960891-83dae900-59d5-11eb-8af8-2121b1248d83.png)

Ludek chodzac sobie po stronie generuje jakies logi, ktore trafiaja do eventhuba i potem (wersja drozsza) do cosmosa lub (wersja tansza) - table storage.
Jezeli ludek kupi bilet to z racji tego, ze wazna jest kolejnosc zgloszen skorzystalbym z service busa. 
Jezeli platnosc sie nie powiedzie to skorzystalbym z eventgrida aby wypchnac jakas wiadomosc mailowa do Ludka. Analogicznie jezeli ludek zapyta o zmiane zasobu.

**Reply by aryll** [reply]
@phaate-ls dzięki za diagram! mam nadzieję, że ośmieli to resztę osób 😄 pozwolisz, że omówimy Twoje rozwiązanie na spotkaniu w czwartek?

**Reply by phaate** [reply]
Pewnie. 
Do czwartku jeszcze to przepisze i poprawnie ponazywam dokladnie czego uzywam.

**Reply by eklime** [reply]
To ja tylko podpowiem, że między Event Grid a Mailem brakuje jeszcze jednego elementu ;)

**Reply by aryll** [reply]
@eklime work in progress - daj się innym pobawić 😃

**Reply by phaate** [reply]
Event Grid jest tylko triggerem, sam niczego nie wykona :P

![image](https://user-images.githubusercontent.com/55980809/105341788-ce2fb600-5bdf-11eb-91ec-13642fb60ee1.png)

**Comment by aryll** [comment]
Szybka przypominajka - widzimy się **dzisiaj o 20** i omawiamy systemy przesyłania komunikatów w Azure!
Nadal czekamy na Wasze diagramy 🤩

**Comment by piotrgwiazda** [comment]
Pamiętajcie, że Event Grid nie wspiera CMK, ani nie gwaratuje lokalizacji
danych (jest pominięty w warunkach umownych z nieznanych przyczyn) zatem
nie wiem jakiego rodzaju dane mogą w ogóle przez niego przechodzić w
kategorii około-RODO i jak pada słowo "payments". Pewnie nie położyłbym tam
nawet identyfikatora płatności. Poza tym gwarancja dostarczenia jest
"zwykle raz".
Event Grid traktowałbym jak dzwonek domofonem i "proszę zejść po paczkę".

czw., 21 sty 2021, 11:58 użytkownik phaate-ls <notifications@github.com>
napisał:

**Reply by aryll** [reply]
@piotrgwiazda w dokumentacji widzę że delivery-guarantee jest 'at-least-once' i działa retry do momentu gdy nie osiągnie się maksymalnej ilości prób które są zdefiniowane.

Mógłbyś podrzucić jakis link związany że jest to "zwykle raz"? bo nie mogę tego znaleźć

(do diagramu w tej chwili się nie odnoszę w ogóle)

**Comment by piotrgwiazda** [comment]
Redundacja w przypadku georeplikacji dotyczy jedynie metadanych,
nieodebrane wiadomości przepadają
https://docs.microsoft.com/en-us/azure/event-grid/geo-disaster-recovery

Ten mechanizm retry z back-off może powodować, że po przywróceniu usługi
nie da się szybko przetworzyć zaległych wiadomości, tylko mogą pojawić dość
póżno:

10 sekund
30 sekund
1 minuta
5 min
10 minut
30 minut
1 godzina
3 godziny
6 godz.

czw., 21 sty 2021, 16:24 użytkownik Agnieszka Cent <notifications@github.com>
napisał:
